### PR TITLE
A working check if local file is installed

### PR DESCRIPTION
**Pull request by Jmainguy**

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @Jmainguy for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

**Comment by drzraf** [comment]
works great, but why not using 
`"dpkg -f %s Package" % deb_file`
which will avoid "parsing" the whole output from `--info`?
If `m.run_command(cmd)` is a shell, subshell (`$( )`) may even be useful to run only once so that's is quicker and less racy.

**Comment by Jmainguy** [comment]
@drzraf Dadgumit thats way easier. Basically because I was unaware of -f until now, this is pretty sweet. I will adjust the PR to use that.

**Comment by Jmainguy** [comment]
@drzraf take a gander now.

**Comment by drzraf** [comment]
code is more clear now. Just tested it a bit with my playbook and it works as expected too.

I didn't caught in the first place and I don't know about official recommendation about command executions policy (like "trying to reduce them or not"), but :
`dpkg -f  %s  Package Version" % deb_file`
would have avoid making two subcommand calls (at the price of going back to line splitting) :)

But personally I don't mind.
Keep up the good work!

**Comment by Jmainguy** [comment]
@bcoca @gregdek looks like this is still waiting to be reviewed since December 17th.

File: packaging/os/apt.py [hunk 1]
```diff
@@ -409,9 +409,21 @@ def install_deb(m, debs, cache, force, install_recommends, dpkg_options):
         try:
             pkg = apt.debfile.DebPackage(deb_file)
 
-            # Check if it's already installed
-            if pkg.compare_to_version_in_cache() == pkg.VERSION_SAME:
-                continue
+            # Check name and version of local file
+            pcmd = "dpkg --field %s Package" % deb_file
+            vcmd = "dpkg --field %s Version" % deb_file
```
Comment (bcoca) [inline packaging/os/apt.py:414]: you should use the find bin function to get dpkg

File: packaging/os/apt.py [hunk 2]
```diff
@@ -409,9 +409,21 @@ def install_deb(m, debs, cache, force, install_recommends, dpkg_options):
...
+            pkg_name = m.run_command(pcmd)[1].strip('\n')
+            pkg_version = m.run_command(vcmd)[1].strip('\n')
```
Comment (bcoca) [inline packaging/os/apt.py:416]: no error checks/trapping?

File: packaging/os/apt.py [hunk 3]
```diff
@@ -409,9 +409,21 @@ def install_deb(m, debs, cache, force, install_recommends, dpkg_options):
...
+            # Check if already installed, or needs upgrading.
+            try:
+                installed_pkg = apt.Cache()[pkg_name]
+                installed_version = installed_pkg.installed.version
+                if package_version_compare(pkg_version, installed_version) <= 0:
+                    # Does not need to upgrade, move on to next package
```
Comment (bcoca) [inline packaging/os/apt.py:422]: what happens when there is a downgrade?

**Comment by gregdek** [comment]
@Jmainguy A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@Jmainguy Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**